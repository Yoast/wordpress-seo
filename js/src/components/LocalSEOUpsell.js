/* External dependencies */
import { __, sprintf } from "@wordpress/i18n";
import PropTypes from "prop-types";
import styled from "styled-components";
import { makeOutboundLink, getDirectionalStyle as getRtlStyle } from "@yoast/helpers";

/* Internal dependencies */
import LocalAssistantImage from "../../../images/local_assistant.svg";

const Container = styled.div`
	display: flex;
	flex-direction: row;
	max-width: 600px;
	margin: 16px 0;
	padding: 20px 28px;
	position: relative;
	z-index: 1;
	&:before {
		content: "";
		background-image: url(${ props => props.backgroundUrl });
		background-size: 100% 100%;
		transform: ${ getRtlStyle( "", "scaleX(-1)" ) };
		position: absolute;
		top: 0;
		left: 0;
		right: 0;
		bottom: 0;
		z-index: -1;
	}
`;

const StyledLocalAssistantImage = styled( LocalAssistantImage )`
	align-self: center;
	width: 100px;
	margin: ${ getRtlStyle( "0 20px 0 0", "0 0 0 20px" ) };
`;

const TextContainer = styled.div`
	align-self: center;
	font-size: 1em;
`;

const Header = styled.h3`
	&& {
		font-size: 1.15em;
		font-weight: bold;
		margin: 4px 0;
	}
`;

const Paragraph = styled.p`
	margin: 4px 0;
`;

const OutboundLink = styled( makeOutboundLink() )`
	margin: 4px 0;
`;

/**
 * Renders the LocalSEOUpsell component.
 *
 * @param {Object} props               The component's props.
 * @param {string} props.url           The url the outbound link goes to.
 * @param {string} props.backgroundUrl The url for the background image of the container.
 *
 * @returns {wp.Element} The rendered LocalSEOUpsell component.
 */
const LocalSEOUpsell = props => {
	const { url, backgroundUrl } = props;

	return (
		<Container backgroundUrl={ backgroundUrl }>
			<StyledLocalAssistantImage />
			<TextContainer>
				<Header>
					{ __( "Serving local customers?", "wordpress-seo" ) }
				</Header>
				<Paragraph>
					{ sprintf(
						/* translators: %s expands to Local SEO */
						__(
							"Truly optimize your site for a local audience with our %s plugin! Optimized address details, opening hours, " +
							"store locator and pickup option!",
							"wordpress-seo"
						),
						"Local SEO"
					) }
				</Paragraph>
				<OutboundLink href={ url }>
					{ sprintf(
						/* translators: %s expands to Local SEO */
						__( "Get the %s plugin now", "wordpress-seo" ),
						"Local SEO"
					) }
				</OutboundLink>
			</TextContainer>
		</Container>
	);
};

LocalSEOUpsell.propTypes = {
	url: PropTypes.string.isRequired,
	backgroundUrl: PropTypes.string.isRequired,
};

export default LocalSEOUpsell;
