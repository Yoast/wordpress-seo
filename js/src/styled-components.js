import * as styledComponents from "../../node_modules/styled-components";

window.yoast = window.yoast || {};
window.yoast.styledComponents = styledComponents;
