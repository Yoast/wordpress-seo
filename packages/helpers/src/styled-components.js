/**
 * Returns a function that returns the correct style based on if the current
 * theme is right to left.
 *
 * This is determined by the `isRtl` property in the styled components theme.
 *
 * @param {string} leftToRightStyle  Style to return if the theme is left to right.
 * @param {string} rightToLeftStyle Style to return if the theme is right to left.
 *
 * @returns {Function} A function that returns the right styled based on the
 *                       theme in the props.
 */
export function getDirectionalStyle( leftToRightStyle, rightToLeftStyle ) {
	return ( props ) => {
		return ( props.theme.isRtl ? rightToLeftStyle : leftToRightStyle );
	};
}
