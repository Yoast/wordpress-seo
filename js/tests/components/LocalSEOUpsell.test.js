import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import LocalSEOUpsell from "../../src/components/LocalSEOUpsell";

describe( "LocalSEOUpsell", () => {
	it( "matches the snapshot", () => {
		const component = renderer.create(
			<LocalSEOUpsell
				url="https://example.com"
				backgroundUrl="https://example.com/image"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "matches the snapshot in right to left mode", () => {
		const component = renderer.create(
			<ThemeProvider theme={ { isRtl: true } }>
				<LocalSEOUpsell
					url="https://example.com"
					backgroundUrl="https://example.com/image"
				/>
			</ThemeProvider>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
