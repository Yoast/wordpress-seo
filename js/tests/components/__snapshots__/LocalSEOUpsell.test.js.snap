// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalSEOUpsell matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  max-width: 600px;
  margin: 16px 0;
  padding: 20px 28px;
  position: relative;
  z-index: 1;
}

.c0:before {
  content: "";
  background-image: url(https://example.com/image);
  background-size: 100% 100%;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 100px;
  margin: 0 20px 0 0;
}

.c2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-size: 1em;
}

.c3.c3 {
  font-size: 1.15em;
  font-weight: bold;
  margin: 4px 0;
}

.c4 {
  margin: 4px 0;
}

.c5 {
  margin: 4px 0;
}

<div
  className="c0"
>
  <test-file-stub
    className="c1"
  />
  <div
    className="c2"
  >
    <h3
      className="c3"
    >
      Serving local customers?
    </h3>
    <p
      className="c4"
    >
      Truly optimize your site for a local audience with our Local SEO plugin! Optimized address details, opening hours, store locator and pickup option!
    </p>
    <a
      className="c5"
      href="https://example.com"
      rel="noopener"
      target="_blank"
    >
      Get the Local SEO plugin now
      <span
        className="A11yNotice-sc-1feyioo-0 gHCoqL"
      >
        (Opens in a new browser tab)
      </span>
    </a>
  </div>
</div>
`;

exports[`LocalSEOUpsell matches the snapshot in right to left mode 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  max-width: 600px;
  margin: 16px 0;
  padding: 20px 28px;
  position: relative;
  z-index: 1;
}

.c0:before {
  content: "";
  background-image: url(https://example.com/image);
  background-size: 100% 100%;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: -1;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 100px;
  margin: 0 0 0 20px;
}

.c2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-size: 1em;
}

.c3.c3 {
  font-size: 1.15em;
  font-weight: bold;
  margin: 4px 0;
}

.c4 {
  margin: 4px 0;
}

.c5 {
  margin: 4px 0;
}

<div
  className="c0"
>
  <test-file-stub
    className="c1"
  />
  <div
    className="c2"
  >
    <h3
      className="c3"
    >
      Serving local customers?
    </h3>
    <p
      className="c4"
    >
      Truly optimize your site for a local audience with our Local SEO plugin! Optimized address details, opening hours, store locator and pickup option!
    </p>
    <a
      className="c5"
      href="https://example.com"
      rel="noopener"
      target="_blank"
    >
      Get the Local SEO plugin now
      <span
        className="A11yNotice-sc-1feyioo-0 gHCoqL"
      >
        (Opens in a new browser tab)
      </span>
    </a>
  </div>
</div>
`;
